### PR TITLE
Creation of All Products Block

### PR DESCRIPTION
**Pull request by mikejolley**
In progress attempt at #898. Currently only implementing the creation of the block, default placeholders if content/products are missing, and inspector settings.

Attributes and inspector controls I anticipate will be shared are in the `assets/js/blocks/products` directory. This block itself is under `assets/js/blocks/products/all-products`. This follows the patterns set by the new reviews blocks.

### CODE REVIEW

File: assets/css/style.scss [hunk 1]
```diff
@@ -1,206 +1 @@
...
-	// Change the title font to match headings.
-	.wc-block-grid__product-title,
-	.wc-block-grid__product-onsale {
-		font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif;
-	}
-	.wc-block-grid__product-onsale {
-		line-height: 1;
-	}
-}
+/* Moved */
```
Comment (Aljullu) [inline assets/css/style.scss]: Should we just remove the file?

File: assets/js/base/components/product-grid/style.scss [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+	.wc-block-grid__product-rating__stars {
+		overflow: hidden;
+		position: relative;
+		width: 5.3em;
+		height: 1.618em;
+		line-height: 1.618;
+		font-size: 1em;
+		font-family: star; /* stylelint-disable-line */
+		font-weight: 400;
+		display: -block;
```
Comment (Aljullu) [inline assets/js/base/components/product-grid/style.scss]: Should this be `inline-block`?
```suggestion
		display: inline-block;
```

File: assets/js/blocks/products/all-products/block.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+		componentId: PropTypes.number,
+	}
+
+	render() {
+		/**
+		 * Todo classes
+		 *
+		 * wp-block-{$this->block_name},
+		 * wc-block-{$this->block_name},
+		 */
```
Comment (Aljullu) [inline assets/js/blocks/products/all-products/block.js]: What does this comment mean?

File: package.json [hunk 4]
```diff
@@ -107,6 +107,7 @@
 	"dependencies": {
 		"@woocommerce/components": "3.2.0",
 		"gridicons": "3.3.1",
+		"react-number-format": "^4.2.0",
```
Comment (Aljullu) [inline package.json]: I wonder if we could achieve the same results with [`@woocommerce/number`](https://www.npmjs.com/package/@woocommerce/number), so we use a dependency that we control.

File: src/Assets.php [hunk 5]
```diff
@@ -114,6 +117,7 @@ public static function get_wc_block_data( $settings ) {
 				'isLargeCatalog'     => $product_counts->publish > 200,
 				'limitTags'          => $tag_count > 100,
 				'hasTags'            => $tag_count > 0,
+				'hasProducts'        => $product_counts->publish > 0,
```
Comment (Aljullu) [inline src/Assets.php]: What would you think about naming this setting as `hasPublishedProducts`? I think it describes better its value.

File: src/RestApi/Controllers/Products.php [hunk 6]
```diff
@@ -107,11 +107,15 @@ public function get_item_permissions_check( $request ) {
 	/**
 	 * Change read permissions to allow author access to this API.
 	 *
-	 * @param array $permissions Array of access permissions.
+	 * @param bool   $permission Permission.
+	 * @param string $context Context of the request.
```
Comment (Aljullu) [inline src/RestApi/Controllers/Products.php]: Minor alignment issue.

File: package.json (hunk 4 above)
Comment (mikejolley) [reply package.json]: Last I checked it added extra dependencies we wish to avoid (lodash)

Comment (Aljullu) [reply package.json]: Good point. I created a PR to remove lodash as a dependency (https://github.com/woocommerce/woocommerce-admin/pull/2977) but I'm ok using `react-number-format` anyway.

**Comment by nerrad** [comment]
closing this in favor of #1032

**Pull request closed without merging**